tools/ferny

```diff
@@ -0,0 +1,26 @@
+#!/bin/sh
```

Comment:
Meh -- can we just change tools/systemd_ctypes to tools/install-submodules ?

Reply:
Meh.

Are you volunteering? :)

At some point I start to wonder if we can remove some of this scripting.  Things like `systemd_ctypes` and `ferny` are actually pretty small and might work fine via the normal git commands.

It's also interesting to note that this works:

```sh
git clone --recurse-submodules='modules/*' https://github.com/cockpit-project/cockpit
```

and avoids the heavy pixel references and node_modules repositories.

Reply:
Yes, it also works fairly well with pixel refs. The problem is that this --recurse-submodules is a magic "unbreak my setup", instead of just being a property of the actual repo (so that `git clone <url>` would just DTRT). Unfortunately git submodules leave a lot to be desired :cry: In other words, we need to get along with non-recursive clones anyway, so we can just as well always check them out ourselves.

I think we could even drop the fancy git-utils.sh machinery for modules/* if it gets in the way -- these are all tiny . If it helps, I can send a PR that renames/generalizes tools/systemd_ctypes first, but it's fairly obvious either way.

Reply:
I'm willing to clean up this duplication in a follow-up. This is really rather trivial, but ok to not block this PR.